### ISSUE TITLE
Cannot use the key "id" in a struct for YYC

### ISSUE THREAD

**Issue by shichen85** [YoYoGames/GameMaker-Bugs#4749]
### Description
The following code snippet can compile for YYC in the 2023 November runtime and 2024.200.0.511, but now throws Compile Error for "assignment to read only variable"

```
m = ds_map_create();
m[?"h"] = "helloworld"

function Cakeframe () constructor {
	static register = function register(is_root=true){
		if is_undefined(id) {
			self[$ "id"] = m[?"h"]
		}
	}
}
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.200.0.495 Runtime v2024.200.0.512

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22000.0

[0b5ae9b1-8b3e-4e33-9dfb-63392d480522](https://api.gamemaker.io/api/github/downloads/0b5ae9b1-8b3e-4e33-9dfb-63392d480522)

**Comment by rwkay** [comment]
Good catch!  I have added a fix for this into 2024.2 - should be in the next Beta.

**Comment by rwkay** [comment]
Fixed in 2024.2

**Comment by sihammill** [comment]
Verified in IDE v2024.200.0.497  Runtime v2024.200.0.514
